lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java

```diff
@@ -158,14 +173,52 @@ WordCase caseOf(char[] word, int length) {
     return null;
   }
 
+  List<char[]> sharpSVariations(char[] word, int length) {
+    if (!dictionary.checkSharpS) return Collections.emptyList();
+
+    Stream<String> result =
+        new Object() {
+          int findSS(int start) {
+            for (int i = start; i < length - 1; i++) {
+              if (word[i] == 's' && word[i + 1] == 's') {
```

Comment:
Is there anything like `ArrayUtil.indexOf(char[] haystack, char[] needle)`?

Reply:
I don't think there is. There is an ancient discussion somewhere on jdk's mailing lists about making String.indexOf(String) more algorithmically efficient but it never made it through - it's hard to beat the naive algorithm on average, common case and if you have to deal with degenerates, it's advised to just roll out your own version.

Reply:
If it were a `String`, I'd use `indexOf`, but it's a `char[]`. Pity :(

Reply:
I like the exponential candidate expansion guard limit (5) -- is this something empirical? I know it's German and its compounds make things more difficult but 5 double 's' still seems like a large limit to me. Is this from hunspell code too?

Reply:
Yes, https://github.com/hunspell/hunspell/blob/master/src/hunspell/hunspell.hxx#L85 and

Reply:
Thanks.